client/src/partials/cash/cash.js

```diff
@@ -97,45 +100,68 @@ function CashController(Cash, Cashboxes, AppCache, Currencies, Modal, $statePara
    */
   function startup() {
 
+    if ((!cache.cashbox || (cache.cashbox && !cache.cashbox.id)) && !cashboxId) {
+
+      /**
+       * first case :
+       * No data in the cache && No data in stateParams
+       */
+      openSelectCashboxModal();
+
+    } else if (cache.cashbox && cache.cashbox.id && !cashboxId) {
+
+      /**
+       * second case :
```

Comment:
These are excellent comments.  I think this code can be improved in a refactor, but this is exactly the functionality we wish to achieve.  Thanks!